packages/local_auth/local_auth_platform_interface/lib/local_auth_platform_interface.dart

```diff
@@ -0,0 +1,91 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:local_auth_platform_interface/default_method_channel_platform.dart';
+import 'package:local_auth_platform_interface/types/auth_options.dart';
+import 'package:local_auth_platform_interface/types/biometric_type.dart';
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
+
+/// The interface that implementations of local_auth must implement.
+///
+/// Platform implementations should extend this class rather than implement it as `local_auth`
+/// does not consider newly added methods to be breaking changes. Extending this class
+/// (using `extends`) ensures that the subclass will get the default implementation, while
+/// platform implementations that `implements` this interface will be broken by newly added
+/// [LocalAuthPlatform] methods.
+abstract class LocalAuthPlatform extends PlatformInterface {
+  /// Constructs a LocalAuthPlatform.
+  LocalAuthPlatform() : super(token: _token);
+
+  static final Object _token = Object();
+
+  static LocalAuthPlatform _instance = DefaultLocalAuthPlatform();
+
+  /// The default instance of [LocalAuthPlatform] to use.
+  ///
+  /// Defaults to [DefaultLocalAuthPlatform].
+  static LocalAuthPlatform get instance => _instance;
+
+  /// Platform-specific implementations should set this with their own
+  /// platform-specific class that extends [LocalAuthPlatform] when they
+  /// register themselves.
+  static set instance(LocalAuthPlatform instance) {
+    PlatformInterface.verifyToken(instance, _token);
+    _instance = instance;
+  }
+
+  /// Authenticates the user with biometrics available on the device while also
+  /// allowing the user to use device authentication - pin, pattern, passcode.
+  ///
+  /// Returns a [Future] holding true, if the user successfully authenticated,
+  /// false otherwise.
+  ///
+  /// [localizedReason] is the message to show to user while prompting them
+  /// for authentication. This is typically along the lines of: 'Please scan
+  /// your finger to access MyApp.'. This must not be empty.
+  ///
+  /// Provide [authStrings] if you want to
+  /// customize messages in the dialogs.
```

Comment:
We should say what the keys are supposed to be here.

Reply:
Actually, looking at the current plugin... how is this intended to work? In the current implementation the keys here are platform-specific. How would a platform-agnostic front-end package call this with the correct things?

Is there a design proposal somewhere for how this will be structured?

Reply:
This is indeed a bit of a weird one. The current plugin has two separate classes for iOS and Android which the `authenticate` method takes as optional parameters. These have been moved to their platform specific packages, but are still used in the app facing package as changing that would be a breaking change. As these classes are not platform agnostic, it doesn't seem right to use them in the platform interface, but that does lead to a situation like this where the keys are unknown. 

I'm not aware of any design proposal, and neither do I know what the right solution would be here.

Reply:
We'll want to deprecate that method and replace it with a new one that doesn't take the specific platforms.

I think the best option here is probably:
- Declare an empty base class for platform strings (`AuthMessages`) in the platform interface.
- Use that in the platform interface and app-facing APIs.
- Make a public subclass in each platform implementation package.
  - (To make moving the existing ones non-breaking, you'll need to re-`export` them for now from the app-facing package, with a TODO to remove it the next time there's a breaking change.)

Then clients would create and pass in the correct one via platform checks if they want to override strings.

Or actually, as a variant that's a bit easier to call: you could make the parameter be an `Iterable<AuthMessages>` instead of an `AuthMessages`, and then clients would just pass all of them in a list instead of needing to do platform checks. Each implementation would filter for only its own subclass within the list.

Reply:
@stuartmorgan Agreed, that sounds like a more sane approach. I've updated all 4 PRs to implement this.